src/FluentMigrator.Extensions.Postgres/Postgres/PostgresExtensions.UsingIndexAlgorithm.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Gist);
+            return expression;
+        }
+
+        public static ICreateIndexNonKeyColumnSyntax UsingGistAlgorithm(this ICreateIndexOnColumnSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Gist);
+            return new CreateIndexExpressionNonKeyBuilder(expression, additionalFeatures);
+        }
+
+        public static ICreateIndexOptionsSyntax UsingSpgistAlgorithm(this ICreateIndexOptionsSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Spgist);
+            return expression;
+        }
+
+        public static ICreateIndexNonKeyColumnSyntax UsingSpgistAlgorithm(this ICreateIndexOnColumnSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Spgist);
+            return new CreateIndexExpressionNonKeyBuilder(expression, additionalFeatures);
+        }
+
+        public static ICreateIndexOptionsSyntax UsingGinAlgorithm(this ICreateIndexOptionsSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Gin);
+            return expression;
+        }
+
+        public static ICreateIndexNonKeyColumnSyntax UsingGinAlgorithm(this ICreateIndexOnColumnSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Gin);
+            return new CreateIndexExpressionNonKeyBuilder(expression, additionalFeatures);
+        }
+
+        public static ICreateIndexOptionsSyntax UsingBrinAlgorithm(this ICreateIndexOptionsSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Brin);
+            return expression;
+        }
+
+        public static ICreateIndexNonKeyColumnSyntax UsingBrinAlgorithm(this ICreateIndexOnColumnSyntax expression)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.UsingIndexAlgorithm(PostgresIndexAlgorithm.Brin);
+            return new CreateIndexExpressionNonKeyBuilder(expression, additionalFeatures);
+        }
+
+        internal static void UsingIndexAlgorithm(this ISupportAdditionalFeatures additionalFeatures, PostgresIndexAlgorithm algorithm)
```

Comment:
Please create a `public` extension method variant of this that:

1. Is named `Using` (so that it auto-completes as `Using(PostgresIndexAlgorithm.Btree)`
2. hangs off `this ICreateIndexOptionsSyntax expression`
3. hangs off `this ICreateIndexOnColumnSyntax expression`

That would be perfect, in terms of what I had in my mind.